tsl/src/compression/api.c

```diff
@@ -199,18 +280,131 @@ restore_autovacuum_on_decompress(Oid uncompressed_hypertable_relid, Oid uncompre
 	}
 }
 
-static void
+Chunk *
+find_chunk_to_merge_into(Hypertable *ht, Chunk *current_chunk)
+{
+	int64 max_chunk_interval, current_chunk_interval = 0, compressed_chunk_interval = 0;
+	const Dimension *time_dim;
+	Chunk *previous_chunk;
+	Point *p;
+	int64 val;
+
+	time_dim = hyperspace_get_open_dimension(ht->space, 0);
+	if (!time_dim)
+		elog(ERROR, "hypertable has no open partitioning dimension");
+
+	if (time_dim->fd.compress_interval_length == 0)
+		return NULL;
+
+	max_chunk_interval = time_dim->fd.compress_interval_length;
+
+	p = ts_point_create(current_chunk->cube->num_slices);
+	for (int i = 0; i < current_chunk->cube->num_slices; i++)
+	{
+		val = current_chunk->cube->slices[i]->fd.range_start;
+		if (current_chunk->cube->slices[i]->fd.dimension_id == time_dim->fd.id)
+		{
+			current_chunk_interval = current_chunk->cube->slices[i]->fd.range_end -
+									 current_chunk->cube->slices[i]->fd.range_start;
+			val--;
+		}
+		p->coordinates[p->num_coords++] = val;
+	}
```

Comment:
```suggestion
  if (current_chunk->cube->slices[i]->fd.dimension_id != time_dim->fd.id)
  {
    elog(ERROR,"time dimension is not primary dimension");
  }
  p->coordinates[p->num_ccords++] = current_chunk->cube->slices[0]->fd.range_start - 1;
	for (int i = 1; i < current_chunk->cube->num_slices; i++)
	{
		p->coordinates[p->num_coords++] = current_chunk->cube->slices[i]->fd.range_start;
	}
```

The time dimension the merging happens on will always be the first dimension so i think this code could be simplified a bit, we can get time val and calculate new value and always copy the rest.